include/boost/geometry/algorithms/detail/is_valid/multipolygon.hpp

```diff
@@ -290,22 +291,22 @@ class is_valid_multipolygon
         // check validity of all polygons ring
         debug_phase::apply(1);
 
-        if (! detail::check_iterator_range
-                  <
-                      per_polygon<VisitPolicy, Strategy>,
-                      false // do not check for empty multipolygon (done above)
-                  >::apply(boost::begin(multipolygon),
-                           boost::end(multipolygon),
-                           per_polygon<VisitPolicy, Strategy>(visitor, strategy)))
+        if (! (boost::end(multipolygon) == std::find_if(
+            boost::begin(multipolygon),
+            boost::end(multipolygon),
+            per_polygon<VisitPolicy, Strategy>(visitor, strategy))))
```

Comment:
Is preserving the old behavior for empty multipolygons important here? Now it's different.

I'd think about calculating `end` once and replacing `!` and `==` with `!=`.